internal/provider/newrelic/provider_integration_test.go

```diff
@@ -168,12 +181,228 @@ func Test_Getting_external_metric_through_proxy(t *testing.T) {
 	})
 }
 
-func newrelicProviderWithMetric(t *testing.T, metric newrelic.Metric) provider.ExternalMetricsProvider {
-	t.Helper()
+// As The server must be run on port 3000 to be run locally as per the new relic client requirements, subtests
+// cannot be run in parallel.
+//
+// For more details check https://github.com/newrelic/newrelic-client-go/blob/main/pkg/region/region_constants.go
+//
+//nolint:funlen,tparallel
+func Test_does_not_hide_backend_errors(t *testing.T) {
+	t.Parallel()
 
-	testEnv := testutil.TestEnv{}
+	ctx := testutil.ContextWithDeadline(t)
+
+	testEnv := testutil.TestEnv{
+		StartKubernetes: true,
+		Region:          "Local",
+	}
 	testEnv.Generate(t)
 
+	url := fmt.Sprintf("%s/apis/external.metrics.k8s.io/v1beta1/namespaces/*/test_metric", testEnv.BaseURL)
+
+	p := newrelicProviderWithMetric(t, newrelic.Metric{
+		Query: testIntegrationQuery,
+	}, &testEnv)
+
+	runAdapter(t, &testEnv, p)
+
+	//nolint:paralleltest
```

Comment:
is this needed? i think having it at func level is enough

Reply:
I added `paralleltest` at func level, but we need to have it as just with `tparallel` the linter still complains. Check https://github.com/newrelic/newrelic-k8s-metrics-adapter/pull/52/commits/0fff43e9a28e9f4545222e9f85adecff3a15c642

Reply:
I added `paralleltest` at func level, but we need to have it as just with `tparallel` the linter still complains. Check 0e6bfd2c81bd54f0fac543e3094fd3f1d445e536